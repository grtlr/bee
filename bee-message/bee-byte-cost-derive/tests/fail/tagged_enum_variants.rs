// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

#![allow(unused_imports, unreachable_patterns)]

use bee_byte_cost::ByteCost;

#[derive(ByteCost)]
enum SingleEnum {
    #[byte_cost(key)]
    SingleKey(u32),
}

fn main() {}
