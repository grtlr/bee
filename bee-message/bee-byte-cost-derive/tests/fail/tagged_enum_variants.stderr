error: weight types cannot be applied to variants of an enum
  --> tests/fail/tagged_enum_variants.rs:11:5
   |
11 |     SingleKey(u32),
   |     ^^^^^^^^^
